ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -1311,7 +1311,7 @@ class JITModBuilder {
             if (IID == Intrinsic::vastart || IID == Intrinsic::vaend ||
                 IID == Intrinsic::smax ||
                 IID == Intrinsic::usub_with_overflow) {
-              if (NewControlPointCall != nullptr)
+              if (NewControlPointCall != nullptr && OutlineCallDepth == 0)
```

Comment:
I wonder if it would be good to have a function `areWeOutlining()` or something. That would make this code easier to read.

Reply:
yeh.

`OutlineCallDepth == 0` is a bit opaque for a casual reader, don't you think?

Reply:
Maybe `willBeOutlined` is a better name?

Reply:
Fixed in [ba877c6](https://github.com/ykjit/yk/pull/568/commits/ba877c6d94bbc299f3dd8d71bc86b37952245c6a)